src/Aspire.Hosting.Azure.KeyVault/AzureKeyVaultResource.cs

```diff
@@ -64,18 +64,27 @@ public override ProvisionableResource AddAsExistingResource(AzureResourceInfrast
     {
         var bicepIdentifier = this.GetBicepIdentifier();
         var resources = infra.GetProvisionableResources();
-        
+
         // Check if a KeyVaultService with the same identifier already exists
         var existingStore = resources.OfType<KeyVaultService>().SingleOrDefault(store => store.BicepIdentifier == bicepIdentifier);
-        
+
         if (existingStore is not null)
         {
             return existingStore;
         }
-        
+
         // Create and add new resource if it doesn't exist
         var store = KeyVaultService.FromExisting(bicepIdentifier);
-        store.Name = NameOutputReference.AsProvisioningParameter(infra);
+
+        if (!TryApplyExistingResourceNameAndScope(
```

Comment:
Do we need to do this for more Azure resources? Or is KeyVault the only one with this problem?

Reply:
https://github.com/dotnet/aspire/issues/10992